pkg/app/master/command/cliparamsfile.go

```diff
@@ -0,0 +1,149 @@
+
+	if err != nil {
+		xc.Out.Error("params.file.param.value.error", err.Error())
+		xc.Out.State("exited",
+			ovars{
+				"exit.code": -1,
+			})
+		xc.Exit(-1)
+	}
+
+}
+
+func validFlagsMap(xc *app.ExecutionContext, validFlags []cli.Flag) map[string]cli.Flag {
+	validFlagsMap := make(map[string]cli.Flag)
+
+	for _, flag := range validFlags {
+		if len(flag.Names()) == 0 {
+			xc.Out.Error("params-file.validate.params", "flag names has no values")
+			xc.Out.State("exited",
+				ovars{
+					"exit.code": -1,
+				})
+			xc.Exit(-1)
+		}
+
+		validFlagsMap[flag.Names()[0]] = flag
+	}
+
+	return validFlagsMap
+}
+
+func flagSetString(flag cli.Flag, paramValue interface{}) (string, error) {
+	paramValueType := reflect.TypeOf(paramValue).Kind()
+
+	switch flag.(type) {
+	case *cli.StringFlag:
+		if paramValueType == reflect.String {
+			return paramValue.(string), nil
+		} else {
+			return "", fmt.Errorf("expected string value found: %v for flag %s", paramValueType, flag.Names()[0])
+		}
+	case *cli.BoolFlag:
+		if paramValueType == reflect.Bool {
+			return strconv.FormatBool(paramValue.(bool)), nil
+		} else {
+			return "", fmt.Errorf("expected boolean value found: %v for flag %s", paramValueType, flag.Names()[0])
+		}
+	case *cli.IntFlag:
+		// JSON Unmarshal unmarshals a JSON number as a Float64 by default
+		if paramValueType == reflect.Int {
+			return strconv.Itoa(paramValue.(int)), nil
+		} else if paramValueType == reflect.Float64 {
+			// Decimal will be truncated out here
+			return strconv.Itoa(int(paramValue.(float64))), nil
+		} else {
+			return "", fmt.Errorf("expected Int value found: %v for flag %s", paramValueType, flag.Names()[0])
+		}
+	case *cli.StringSliceFlag:
+		// Parse both string slice or string values
+		if paramValueType == reflect.String {
+			return paramValue.(string), nil
+		} else if paramValueType == reflect.Slice {
+			var values []string
+			slice := reflect.ValueOf(paramValue)
+			for i := 0; i < slice.Len(); i++ {
+				values = append(values, fmt.Sprintf("%v", slice.Index(i).Interface()))
+			}
+			return strings.Join(values, ","), nil
+		} else {
+			return "", fmt.Errorf("expected string or string slice value found: %v for flag %s", paramValueType, flag.Names()[0])
+		}
+	default:
+		return "", errors.New("flag type not supported by params file")
+	}
+}
+
+func setFileParams(ctx *cli.Context, xc *app.ExecutionContext, params map[string]interface{}, validFlags []cli.Flag) error {
+	validFlagsMap := validFlagsMap(xc, validFlags)
+
+	for fileParamKey, fileParamValue := range params {
+
+		if flag, ok := validFlagsMap[fileParamKey]; !ok {
+			return fmt.Errorf("the command params file contains an invalid flag - %s", fileParamKey)
+		} else {
+			setValue, err := flagSetString(flag, fileParamValue)
+			if err != nil {
+				return err
+			}
+
+			if ctx.IsSet(fileParamKey) {
+				xc.Out.Info("command.params.file",
+					ovars{
+						"message": "updating already set value from params file",
```

Comment:
Unsure what the expected behaviour should be here for setting a flag value that has already been set.
Currently it just logs to the user if a flag is set that is updating an existing flag that has been set. 
I say update because it depends on the type of flag being used, for example if its a string it will just replace the previous string value with the value from the JSON file, but if its a string slice flag e.g a tag, it  that the cli lib will append the value in the JSON to the existing values.

Reply:
that's reasonable